projects/hathor-wallet-headless/api-docs-ci.md

```diff
@@ -0,0 +1,63 @@
+# Summary
+This feature automates the process of uploading the updated API Docs for the Headless Wallet, while giving the developer a preview of the contents, error checking and linting.
+
+A script is made available for the developer to make all the necessary checks locally and a GitHub Workflow is implemented to:
+- Validate and lint the `api-docs.js` file on every PR to the `dev` branch
+- Deploy the updated docs on every new successful release
+
+# Motivation
+Currently this upload is made manually by a team member, extracting data from the `src/api-docs.js` file and processing it to finally upload its contents to the [Headless Hathor Wallet API](https://wallet-headless.docs.hathor.network/) site.
+
+# Guide-level explanation
+### JS to JSON conversion
+The platform that manages the OpenAPI Documentation static website, [Redoc](https://github.com/Redocly/redoc), only interacts with `.json` files. So we need to convert our `src/api-docs.js` file into a `.json` one whenever we want to use this platform.
+
+A script should be built to make this conversion on the `scripts` folder, outputting the results to the `tmp/` folder. All following steps will be executed on this converted file. Some specific conditions to execute:
+- Remove the `securitySchemes` from the `components` property
+- Ensure the `default` property normally exported by the javascript `.toJSON()` is not present on the output json
+
+This script will be referred to as `scripts/convert-docs.js` from now on.
+
+### Linting and Preview
+A developer should be able to run a `npm run docs` command with three outcomes:
+- A validation check should be executed on the generated docs, alerting any errors found
+- A lint should be executed on the generated docs, displaying its analysis results
+- A local server should be fired, allowing the developer to visualize the results of the doc changes
+
+This would be an automated version of manually executing the following commands:
```

Comment:
Maybe we should make sure to remove the `tmp/api-docs.json` file at the end of the process.

Reply:
I see no negative consequence of having this file lingering at the `tmp` folder:
- On the local dev environment it could be reused by the `preview-docs` command
- On the CI environment, it would be destroyed as soon as the workflow is over

However I think it's a good practice to remove the tmp files if we have no explicit reason to keep it. :+1: 
Changed it on [9d4de6e](https://github.com/HathorNetwork/rfcs/pull/56/commits/9d4de6ebeee1a96b1e82876e2abbe7c127db52e9)